libraries/chain/include/golos/chain/steem_object_types.hpp

```diff
@@ -70,7 +70,9 @@ namespace golos { namespace chain {
             vesting_delegation_expiration_object_type,
             account_metadata_object_type,
             proposal_object_type,
-            required_approval_object_type
+            required_approval_object_type,
+            // serialize state
+            _serialize_reputation_type
```

Comment:
it is safer to enumerate all id types in same enum. prevents conflicts of ids

Reply:
not in this case. `_serialize_reputation_type` is required **only** in serialize_state, it's not part of golos core indexes. actually, it's implementation details of serialize_state.

btw, why define new id? is here problems with existing one? https://github.com/GolosChain/golos/blob/aac31d14ec45d400f083a593cd9d37ffbefbe209/plugins/follow/include/golos/plugins/follow/follow_objects.hpp#L28

Reply:
look in create genesis tool. in this tool we have only single type-id enum. thats why i cannot use type-id from plugins separate enums.

actually, we can add new **core** object types in future and forget change constant in separate file and it will overlap

Reply:
there is no problem to add enum element with custom value to create-genesis

Reply:
ok, there is variant, which adds some restrictions. but it's still better to write workaround for this in create-genesis side, instead of adding new "fake" id to core

Reply:
> actually, we can add new core object types in future and forget change constant in separate file and it will overlap

it's not a problem. if we don't need to serialize new types, the old values will work perfectly. if we need, then this virtual id can be moved to the end of enum, and "overlap" should be fixed anyway

Reply:
yes, this is variant, also near is enum but it is not used at all...

i dislike changes in core when they complicating work, but like them when we are simplifying it

Reply:
we can add storing of new objects but forget update constant if it is in another file.

Reply:
removed from core but added comment